### PR TITLE
(CONT-808) Ruby 3 / Puppet 8 Support

### PR DESCRIPTION
**Pull request by chelnak**
This PR adds support for Ruby3x and Puppet 8

### CODE REVIEW

File: lib/rspec-puppet/example.rb
Comment (ekohl) [inline lib/rspec-puppet/example.rb:16]: I think you can also drop support for RSpec 2 instead of changing the code below.

File: lib/rspec-puppet/example/function_example_group.rb
Comment (ekohl) [inline lib/rspec-puppet/example/function_example_group.rb:111]: Perhaps just delete commented code?

**Review by ekohl — COMMENTED** [review]
I'm wondering if it's better to first drop Puppet 2 and 3 support from the codebase before doing RuboCop changes.

**Comment by chelnak** [comment]
@ekohl All valid suggestions here, thank you 😄 .

I think ultimately it's going to come down to time, for me at least.

I'd be happy to work through PRs from yourself / other community members though. This is something I've just asked about internally as a way to get rspec-puppet rolling again.

**Review by GSPatton — APPROVED** [review]
looks good to me. Will wait for another review

**Review by LukasAud — APPROVED** [review]
LGTM

**Pull request merged**
